src/Microsoft.AspNetCore.Authentication.Facebook/FacebookExtensions.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Authentication.Facebook;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class FacebookAuthenticationOptionsExtensions
+    {
+        /// <summary>
+        /// Adds facebook authentication with options bound against the "Facebook" section 
+        /// from the IConfiguration in the service container.
+        /// </summary>
+        /// <param name="services"></param>
+        /// <returns></returns>
+        public static IServiceCollection AddFacebookAuthentication(this IServiceCollection services)
+            => services.AddFacebookAuthentication(FacebookDefaults.AuthenticationScheme, _ => { });
+
+        public static IServiceCollection AddFacebookAuthentication(this IServiceCollection services, Action<FacebookOptions> configureOptions) 
+            => services.AddFacebookAuthentication(FacebookDefaults.AuthenticationScheme, configureOptions);
+
+        public static IServiceCollection AddFacebookAuthentication(this IServiceCollection services, string authenticationScheme, Action<FacebookOptions> configureOptions)
+        {
+            services.AddSingleton<IConfigureOptions<FacebookOptions>, FacebookConfigureOptions>();
```

Comment:
> specially since it's not consistent (cookies and OAuth2 don't use it)

But OIDC uses it, hem.

It's even worse than what I imagined:

```
// Uses the OpenIdConnect node resolved from IConfiguration.
services.AddOpenIdConnectAuthentication();

// Uses the configuration delegate... PLUS the OpenIdConnect node resolved from IConfiguration (instead of custom-scheme).
services.AddOpenIdConnectAuthentication("custom-scheme", options => { ... });